src/scripts/debugEmailParsing.ts

```diff
@@ -0,0 +1,42 @@
+import assert from "assert";
+import { convert } from "html-to-text";
+import { Source, simpleParser } from "mailparser";
+
+import {
+  Event,
+  ExtractFromEmailResult,
+  NonEmptyArray,
+  extractFromEmail
+} from "../llm/emailToEvents.js";
+import { isDormspam } from "../emailToEvents.js";
+
+/**
+ * Print some debugging information (whether the message was parsed as dormspam),
+ * then parse the email to extract the events and print this.
+ *
+ * @param message The raw email content, as a Buffer or string.
+ * @returns A list of events
+ */
+export async function debugEmailToEvents(messageSource: Source): Promise<NonEmptyArray<Event>> {
+  const parsed = await simpleParser(messageSource, {
+    skipImageLinks: true,
+    skipHtmlToText: false
+  });
+  assert(parsed.html);
```

Comment:
It would be nice if we could parse plain text files. That would make debugging "why this email is not working" even easier, because the work flow changes from "go to afs and remote copy from saved_emails" to just copying the text down from the email client.
```
Subject: [Macgregor] HSC Hanuman Jayanthi Celebrations!
Body:
Rama Navami +

Hanuman Jayanthi +

Navaratri Celebrations!



on April 8th at 5pm @ the Chapel



As part of the celebrations, we'll have an Open Mandir, Poojas done in diverse Hindu Traditions, and Kirtans followed by a Delicious Banquet!  Details are as below



  *   What: Rama Navami + Hanuman Jayanthi + Navaratri Celebrations!
  *   When: April 8th @ 5pm (Saturday)
  *   Where: MIT Chapel


For more up-to-date news about events from MIT HSC, please forward this email to other people who you think might be interested and check out our website<https://web.mit.edu/hsc/www/> ​(https://web.mit.edu/hsc/www/) ​for more events and info about us!





Lovingly,

The Entire HSC Board

bcc'ed to dorms, blue for bc-talk


[cid:fe921cdf-8a7e-4f28-aa49-2c45ef4a380d]
[cid:124f8e71-60b4-4090-acbf-ad0a00061e4c]
[cid:8883089e-d025-4fef-bed6-507b8669cd1c]

_______________________________________________
MacGregor mailing list
MacGregor@mit.edu
https://mailman.mit.edu/mailman/listinfo/macgregor
```

Reply:
Copying that to stdin might actually just work, I should test that

Reply:
I think I'll just merge and address that someday else. But you can just paste that into notepad and save the file. Or you can right click the email and hit download. I do agree that the workflow could be improved though